modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/ParameterGroupArgumentResolver.java

```diff
@@ -76,10 +76,9 @@ public T resolve(OperationContext operationContext)
             T group = type.newInstance();
             for (Field parameterField : parameterFields)
             {
-                Object value = operationContext.getParameter(getAlias(parameterField));
-                if (value != null)
+                if (operationContext.hasParameter(getAlias(parameterField)))
                 {
-                    parameterField.set(group, value);
+                    parameterField.set(group, operationContext.getParameter(getAlias(parameterField)));
```

Comment:
the code still looks weird. I took a second look an realised I was wrong. Your original commit was indeed equivalent to the original behaviour, so this second version should be rolled back. That been said, if your original commit is equivalent, then why to change it at all?